sparse/src/KokkosSparse_csr2csc.hpp

```diff
@@ -0,0 +1,134 @@
+
+#ifndef _KOKKOSSPARSE_CSR2CSC_HPP
+#define _KOKKOSSPARSE_CSR2CSC_HPP
+namespace KokkosSparse {
+namespace Impl {
+template <class OrdinalType, class SizeType, class ValViewType,
+          class RowMapViewType, class ColIdViewType>
+class Csr2Csc {
+ private:
+  using CscST             = typename ValViewType::value_type;
+  using CscOT             = OrdinalType;
+  using CscET             = typename ValViewType::execution_space;
+  using CscMT             = void;
+  using CscSzT            = SizeType;
+  using CscType           = CcsMatrix<CscST, CscOT, CscET, CscMT, CscSzT>;
+  using CscValsViewType   = typename CscType::values_type;
+  using CscColMapViewType = typename CscType::col_map_type::non_const_type;
+  using CscRowIdViewType  = typename CscType::index_type;
+
+  OrdinalType __nrows;
+  OrdinalType __ncols;
+  SizeType __nnz;
+  ValViewType __vals;
+  RowMapViewType __row_map;
+  ColIdViewType __col_ids;
+
+  CscValsViewType __csc_vals;
+  CscColMapViewType __csc_col_map;
+  CscRowIdViewType __csc_row_ids;
+
+ public:
+  Csr2Csc(OrdinalType nrows, OrdinalType ncols, SizeType nnz, ValViewType vals,
+          RowMapViewType row_map, ColIdViewType col_ids)
+      : __nrows(nrows),
+        __ncols(ncols),
+        __nnz(nnz),
+        __vals(vals),
+        __row_map(row_map),
+        __col_ids(col_ids) {
+    __csc_vals = CscValsViewType(
+        Kokkos::view_alloc(Kokkos::WithoutInitializing, "__csc_vals"), nnz);
+    __csc_col_map =
+        CscColMapViewType(Kokkos::view_alloc("__csc_col_map"), ncols + 1);
+    __csc_row_ids = CscRowIdViewType(
+        Kokkos::view_alloc(Kokkos::WithoutInitializing, "__csc_row_ids"), nnz);
+
+    KokkosSparse::Impl::transpose_matrix<
+        RowMapViewType, ColIdViewType, ValViewType, CscColMapViewType,
+        CscRowIdViewType, CscValsViewType, CscColMapViewType, CscET>(
+        __nrows, __ncols, __row_map, __col_ids, __vals, __csc_col_map,
+        __csc_row_ids, __csc_vals);
+  }
+
+  CscType get_cscMat() {
+    return CscType("csr2csc", __nrows, __ncols, __nnz, __csc_vals,
+                   __csc_col_map, __csc_row_ids);
+  }
+};
+}  // namespace Impl
+///
+/// \brief Converts a csr matrix to a CscMatrix.
+/// \tparam OrdinalType The view value type associated with the RowIdViewType
+/// \tparam SizeType The type of nnz
+/// \tparam ValViewType The values view type
+/// \tparam RowMapViewType The column map view type
+/// \tparam ColIdViewType The row ids view type
+/// \param nrows The number of rows in the csr matrix
+/// \param ncols The number of columns in the csr matrix
+/// \param nnz The number of non-zeros in the csr matrix
+/// \param vals The values view of the csr matrix
+/// \param col_ids The col ids view of the csr matrix
+/// \param row_map The row map view of the csr matrix
+/// \return A KokkosSparse::CcsMatrix.
+///
+/// \note: in KokkosKernels sparse code, adj stands for adjacency list
+//   and here we're passing in a csr matrix with xadj=col_ids and adj=row_map.
```

Comment:
```suggestion
//   and here we're passing in a csr matrix with xadj=row_map and adj=col_ids.
```

In the old naming scheme xadj was always the offsets/ptrs, and adj was the entries/indices.